site/en/docs/extensions/reference/userscripts/index.md

```diff
@@ -0,0 +1,113 @@
+    <figure>
+    {% Img src="image/sQ51XsLqKMgSQMCZjIN0B7hlBO02/3gJmpGWMuvsOk9Jaj2NJ.png", alt="Extensions page",
+    width="400", height="183",  class="screenshot" %}
+
+      <figcaption>
+      Extensions page (chrome://extensions)
+      </figcaption>
+    </figure>
+
+You can determine if developer mode is enabled by checking whether `chrome.userScripts` throws an error. For example:
+
+```javascript
+function isUserScriptsAvailable() {
+  try {
+    // Property access which throws if developer mode is not enabled.
+    chrome.userScripts;
+    return true;
+  } catch {
+    // Not available.
+    return false;
+  }
+}
+```
+
+```javascript
+function isUserScriptsAvailable() {
+  try {
+    // Property access which throws if developer mode is not enabled.
+    chrome.userScripts;
+    return true;
+  } catch {
+    // Not available.
+    return false;
+  }
+}
+```
+
+### Work in isolated worlds
+
+Both user and content scripts can run in an isolated world or in the main world. An isolated world is an execution environment that isn't accessible to a host page or other extensions. This lets a user script change its JavaScript environment without affecting the host page or other extensions' user and content scripts. Conversely, user scripts (and content scripts) are not visible to the host page or the user and content scripts of other extensions. Scripts running in the main world are accessible to host pages and other extensions and are visible to host pages and to other extensions. To select the world, pass `"USER_SCRIPT"` or `"MAIN"` when calling [`userScripts.register()`](#method-register).
+
+To configure a [content security policy](https://developer.mozilla.org/docs/Web/HTTP/CSP) for whichever world you specify, call `userScripts.configureWorld()` as shown below.
+
+```javascript
+chrome.userScripts.configureWorld({
+  csp: "script-src 'self'"
+});
+```
+
+### Messaging
+
+Like content scripts and offscreen documents, user scripts can communicate with other parts of an extension using [messaging](/docs/extensions/mv3/messaging/). Unlike other types of messages, user script messages use their own methods: `runtime.onUserScriptMessage` and `chrome.runtime.onUserScriptConnect` events.
+
+Before sending a message, you must call `configureWorld()` with the `messaging` argument set to `true`. Note that both the `csp` and `messaging` arguments can be passed at the same time.
+
+```javascript
+chrome.userScripts.configureWorld({
+  messaging: true
+});
+```
+
+### Extension updates
+
+User scripts are cleared when an extension updates. You can add them back running code via the [`runtime.onInstalled`](/docs/extensions/reference/runtime/#event-onInstalled) event handler in the extension service worker. Respond only to the [`"update"` reason](/docs/extensions/reference/runtime/#type-OnInstalledReason:~:text=as%20an%20installation.-,%22update%22,-Specifies%20the%20event) passed to the event callback.
+
+## Examples
```

Comment:
Is the update API already live? Should we provide a sample?

Reply:
Boilerplate heading. Below it is our (so far) one sample, Register a script. Let me add some text to resolve that.